control-statistics/src/main/java/fi/nls/oskari/control/statistics/user/DeleteIndicatorHandler.java

```diff
@@ -0,0 +1,77 @@
+package fi.nls.oskari.control.statistics.user;
+
+import fi.nls.oskari.annotation.OskariActionRoute;
+import fi.nls.oskari.cache.JedisManager;
+import fi.nls.oskari.control.*;
+import fi.nls.oskari.control.statistics.StatisticsHelper;
+import fi.nls.oskari.control.statistics.data.StatisticalIndicator;
+import fi.nls.oskari.control.statistics.plugins.StatisticalDatasourcePlugin;
+import fi.nls.oskari.control.statistics.plugins.StatisticalDatasourcePluginManager;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.service.OskariComponentManager;
+import fi.nls.oskari.util.JSONHelper;
+import fi.nls.oskari.util.ResponseHelper;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.oskari.statistics.user.StatisticalIndicatorService;
+
+/**
+ * Deletes indicator that the user has previously saved.
+ * Only allows deletion of the users own indicators
+ */
+@OskariActionRoute("DeleteIndicator")
+public class DeleteIndicatorHandler extends ActionHandler {
+    private static final Logger LOG = LogFactory.getLogger(DeleteIndicatorHandler.class);
+    private StatisticalIndicatorService indicatorService;
+
+    @Override
+    public void init() {
+        super.init();
+        if (indicatorService == null) {
+            indicatorService = OskariComponentManager.getComponentOfType(StatisticalIndicatorService.class);
+        }
+    }
+
+    public void handleAction(ActionParameters params) throws ActionException {
+        // user indicators are user content so deleting one requires to be logged in
+        params.requireLoggedInUser();
+
+        int datasourceId = params.getRequiredParamInt(StatisticsHelper.PARAM_DATASOURCE_ID);
+        int id = params.getRequiredParamInt(ActionConstants.PARAM_ID);
+        StatisticalIndicator ind = indicatorService.findById(id, params.getUser().getId());
+        if(ind == null) {
+            // or might not be the owner
+            throw new ActionDeniedException("Unknown indicator/not the owner: " + id );
+        }
+        JSONObject selectors = params.getHttpParamAsJSON(StatisticsHelper.PARAM_SELECTORS);
+        if(selectors != null) {
+            // if selectors present, regionset is also required
+            int regionset = params.getRequiredParamInt("regionset");
+            try {
+                indicatorService.deleteIndicatorData(id, regionset, selectors.getInt("year"));
+            } catch (JSONException ex) {
+                throw new ActionParamsException("Year was not part of the selectors parameter");
```

Comment:
This caused a moment of confusion. I thought to myself: why would `StatisticalIndicatorService#deleteIndicatorData(...)` throw a JSONException and how could one determine that the error must have been due to the selectors not having a year parameter. Then I understood that it is the `selectors.getInt("year")` that's being `try`'ed here.